pkgs/development/compilers/opensycl/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib, clang15Stdenv, fetchFromGitHub, llvmPackages_15, lld_15, llvmPackages_rocm, rocm-device-libs, python3, rocm-runtime, cmake, boost, libxml2, libffi, makeWrapper, hip, rocmSupport ? false }:
+
+clang15Stdenv.mkDerivation {
+  pname = "OpenSYCL";
+  version = "0.9.4";
+  src = fetchFromGitHub {
+    owner = "OpenSYCL";
+    repo = "OpenSYCL";
+    rev = "v0.9.4";
+    sha256 = "sha256-5YkuUOAnvoAD5xDKxKMPq0B7+1pb6hVisPAhs0Za1ls=";
+  };
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+  ];
+  buildInputs = [
+    libxml2
+    libffi
+    boost
+    llvmPackages_15.openmp
+    llvmPackages_15.libclang.dev
+    llvmPackages_15.llvm
+  ] ++ lib.optionals rocmSupport [
+    hip
+    rocm-runtime
+  ];
+
+  # opensycl makes use of clangs internal headers. It's cmake does not successfully discover them automatically on nixos, so we supply the path manually
+  cmakeFlags = [
+    "-DCLANG_INCLUDE_PATH=${llvmPackages_15.libclang.dev}/include"
+  ];
+
+  # one could add --hipsycl-targets='omp;hip' here, but I guess this is best left to the user
```

Comment:
What does `--hipsycl-targets='omp;hip'` do and how would you suggest a user could add that? Btw most users probably won't find it if it is a comment in a package definition, you could set the meta.longDescription attribute and describe how a user could use this.

Reply:
This is an option for opensycl. It tells the compiler to build for both the CPU and HIP targets. I initially thought, since the user explicitely asks for rocm support, we could set this automatically, but theoretically, one could just as well build with openmp only, or only with HIP. I guess adding this here would be like adding "-O2" to the gcc package, such that everyone always compiles with "-O2". In a similar argument, I would also not add it to the (long) description, as this knowledge is not specific to nix at all.